packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,400 @@
+  private contractCache: Map<string, Buffer>
+  private storageTries: { [key: string]: Trie }
+  private externallyRetrievedStorageKeys: Map<string, Map<string, boolean>>
+  private blockTag: string
+  private root: Buffer | undefined
+  private trie: Trie
+
+  constructor(opts: EthersStateManagerOpts) {
+    super({})
+    this.trie = new Trie({ useKeyHashing: true })
+    this.storageTries = {}
+    this.provider = opts.provider
+    if (typeof opts.blockTag === 'bigint') {
+      this.blockTag = bigIntToHex(opts.blockTag)
+    } else {
+      this.blockTag = opts.blockTag ?? 'latest'
+    }
+
+    this.contractCache = new Map()
+    this.externallyRetrievedStorageKeys = new Map<string, Map<string, boolean>>()
+
+    this._cache = new Cache({
+      getCb: (address) => this.getAccountFromProvider(address),
+      putCb: async (keyBuf, accountRlp) => {
+        await this.trie.put(keyBuf, accountRlp)
+      },
+      deleteCb: async (keyBuf) => {
+        await this.trie.del(keyBuf)
+      },
+    })
+  }
+
+  copy(): EthersStateManager {
+    return new EthersStateManager({ provider: this.provider })
+  }
+
+  /**
+   * Gets the code corresponding to the provided `address`.
+   * @param address - Address to get the `code` for
+   * @returns {Promise<Buffer>} -  Resolves with the code corresponding to the provided address.
+   * Returns an empty `Buffer` if the account has no associated code.
+   */
+  async getContractCode(address: Address): Promise<Buffer> {
+    const code = await this.provider.getCode(address.toString(), this.blockTag)
+    const codeBuffer = toBuffer(code)
+    this.contractCache.set(address.toString(), codeBuffer)
+    return codeBuffer
+  }
+
+  /**
+   * Adds `value` to the state trie as code, and sets `codeHash` on the account
+   * corresponding to `address` to reference this.
+   * @param address - Address of the `account` to add the `code` for
+   * @param value - The value of the `code`
+   */
+  async putContractCode(address: Address, value: Buffer): Promise<void> {
+    // Store contract code in the cache
+    this.contractCache.set(address.toString(), value)
+  }
+
+  /**
+   * Gets the storage value associated with the provided `address` and `key`. This method returns
+   * the shortest representation of the stored value.
+   * @param address -  Address of the account to get the storage for
+   * @param key - Key in the account's storage to get the value for. Must be 32 bytes long.
+   * @returns {Buffer} - The storage value for the account
+   * corresponding to the provided address at the provided key.
+   * If this does not exist an empty `Buffer` is returned.
+   */
+  async getContractStorage(address: Address, key: Buffer): Promise<Buffer> {
+    // Retrieve storage slot from provider if not found in cache
+    await this.getContractStorageFromProvider(address, key)
+
+    const storageTrie = await this._getStorageTrie(address)
+    const foundValue = await storageTrie.get(key) //@ts-ignore
```

Comment:
```suggestion
    const foundValue = await storageTrie.get(key)
```
Doesn't seem to error locally if I remove this